forui/lib/src/widgets/select_group/select_group_item.dart

```diff
@@ -1,136 +1,270 @@
 import 'package:flutter/foundation.dart';
 import 'package:flutter/widgets.dart';
 
+import 'package:meta/meta.dart';
+
 import 'package:forui/forui.dart';
 
-/// An item that represents a selection in a [FSelectGroup].
-class FSelectGroupItem<T> with Diagnosticable {
-  /// The value of the item.
-  T value;
+@internal
+class FSelectGroupItemData<T> extends InheritedWidget {
```

Comment:
I think it might be overkill to add an `InheritedWidget` here. We could easily just pass the style one layer down and remove one layer of abstraction.

Reply:
Personally I think it's fine. Using an InheritedWidget allows us to avoid rebuilding all children each time a button is clicked.